### ISSUE TITLE
ages

### ISSUE THREAD

**Issue by dweisz** [dfm/python-fsps#4]
For a variety of young input ages (e.g., <=0.05 Gyr, although it might extend beyond this) I keep getting the following errors:

COMPSP ERROR, maxtime too small:   10964781.961431850  
chex 8:07pm > python weisz.py
 COMPSP ERROR, maxtime too small:   13803842.646028839  
chex 8:08pm > python weisz.py
 COMPSP ERROR, maxtime too small:   10000000.000000000  
chex 8:08pm > python weisz.py
 COMPSP ERROR, maxtime too small:   50000000.000000000     

I think the offenders are sp.get_mags and sp.get_spectrum().

**Comment by Nolaa** [comment]
I have the exact same problem as @dweisz pointed out in above post and I have no idea how to fix that.

**Comment by bd-j** [comment]
This is an issue with FSPS, not python-FSPS.  When tage is explicitly set it cannot be less than 0.1 Gyr because of stability issues.  If you really need younger ages you can set tage=0 and the output will be for all ages in the isochrones, including ages younger than 0.1 Gyr, which you can then interpolate

**Comment by Nolaa** [comment]
The thing is I'm using sfh=2 and the time in the sample file included in data directory "sfh.dat" ranges from 0.96- 13 Gyr and I think the time shouldn't be the problem. Here is part of my code:
```
sp = fsps.StellarPopulation( sfh=2, imf_type=1, sf_start=0.9)
#sfh_filename='sfh.dat'

wave, spec = sp.get_spectrum(tage=13)
```

**Comment by bd-j** [comment]
sfh=2 is not supported in python-FSPS (see #5) You have to use sfh=0 and do your own interpolations, or stick to the other parametric SFHs (1 and 4)

**Comment by rrgupta** [comment]
I think the more general issue is that simply calling `sp.get_mags()` works fine in the sense that it will output mags at all ages, even if they are ridiculous values (e.g., 142.6342352554833 or 99.0). However, as soon as you add the option `tage` to `get_mags`, which I would think would just interpolate from the output of `sp.get_mags()`, you start to encounter problems such as those noted above. I've also encountered problems if I ask for mags for a `tage` that is less than `sf_start`:
```
>>> sdss_bands = fsps.find_filter('sdss')
>>> sp = fsps.StellarPopulation(sfh=4, tau=5.78, zmet=10, sf_start=3.5)
>>> sp.get_mags(bands=sdss_bands, tage=2.0)
COMPSP ERROR: sf_start>maxtime  stopping...
```
When these COMSP ERRORs occur, the program not only crashes but kicks me out of python, too. Is it possible to make `get_mags()` well-behaved when given `tage` options that fall within the range of the isochrones? Having it output nonphysical values is still preferable to crashing I think, and then it's just "user beware." Or even having a python error or warning to protect against the COMPSP ERROR would be useful. Thanks!

**Comment by bd-j** [comment]
we could add a check in the python wrapper, though it will add overhead.

More generally when you set `tage` its not just an interpolation between the isochron values - the necessary integrals are recomputed exactly for the requested `tage`.  This is important for SFHs that change discontinuously between isochrone ages, e.g. because of non-zero `sf_start` or `sf_trunc`.

**Comment by flyboys** [comment]
@bd-j  How can we get a CSP.out.mags file as a function of log Age , log Lbol , log mass , log sfr  similar to  the files  in fsps/OUTPUTS   ? should i use both get_mags() and set_tabular_sfh() functions to accomlpish similiar results to fsps/OUTPUTS  /mags ?

**Comment by bd-j** [comment]
@flyboys that is correct, please see documentation for `get_mags()`.

**Comment by bd-j** [comment]
I am closing this issue as the original problem was addressed by cconroy20/fsps#26
